### PR TITLE
Move dependencies for rueidiscompat to go.mod

### PR DESCRIPTION
**Pull request by DestroyerAlpha**
https://github.com/redis/rueidis/issues/466
Solves this issue

### CODE REVIEW

**Comment by rueian** [comment]
Hi @DestroyerAlpha, these changes look great. Many thanks!

But the changes in the `rueidiscompat/hscan_test.go` seem to be unnecessary. Could you revert it?

**Comment by DestroyerAlpha** [comment]
Hey @rueian, Are you saying we do not need to replace global imports in hscan_test.go?

**Comment by rueian** [comment]
> Hey @rueian, Are you saying we do not need to replace global imports in hscan_test.go?

Yes, can we just keep it untouched? Or will it be broken if we don’t replace global imports?

**Comment by DestroyerAlpha** [comment]
ginkgo moved to an indirect dependency, until I changed all imports to explicit imports.

**Comment by rueian** [comment]
> ginkgo moved to an indirect dependency, until I changed all imports to explicit imports.

Hi @DestroyerAlpha, so what made you want to change them to explicit imports?

**Comment by DestroyerAlpha** [comment]
To remove ginkgo as a dependency from redis/rueidis altogether and not just present as an indirect dependency

**Comment by DestroyerAlpha** [comment]
@rueian Made changes. Not sure why it didn't work the first time around

File: go.sum [hunk 1]
```diff
@@ -17,30 +11,18 @@ github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
...
-golang.org/x/net v0.17.0 h1:pVaXccu2ozPjCXewfr1S7xza/zcXTity9cCdXQYSjIM=
-golang.org/x/net v0.17.0/go.mod h1:NxSsAGuq816PNPmqtQdLE42eU2Fs7NoRIZrHJAlaCOE=
-golang.org/x/sys v0.14.0 h1:Vz7Qs629MkJkGyHxUlRHizWJRG2j8fbQKjELVSNhy7Q=
-golang.org/x/sys v0.14.0/go.mod h1:/VUhepiaJMQUp4+oa/7Zr1D23ma6VTLIYjOOTFZPUcA=
-golang.org/x/text v0.13.0 h1:ablQoSUd0tRdKxZewP80B+BaqeKJuVhuRxj/dkrun3k=
-golang.org/x/text v0.13.0/go.mod h1:TvPlkZtksWOMsz7fbANvkp4WM8x/WCo/om8BMLbz+aE=
-golang.org/x/tools v0.12.0 h1:YW6HUoUmYBpwSgyaGaZq1fHjrBjX1rlpZ54T6mu2kss=
-golang.org/x/tools v0.12.0/go.mod h1:Sc0INKfu04TlqNoRA1hgpFZbhYXHPr4V5DzpSBTPqQM=
-google.golang.org/protobuf v1.28.0 h1:w43yiav+6bVFTBQFZX0r7ipe9JQ1QsbMgHwbBziscLw=
+github.com/onsi/ginkgo/v2 v2.15.0 h1:79HwNRBAZHOEwrczrgSOPy+eFTTlIGELKy5as+ClttY=
```
Comment (rueian) [inline go.sum:14]:
```suggestion
```

Thanks! But this line looks wired. Could you try to remove the go.sum and run `go mod tidy` again?

**Comment by rueian** [comment]
Closed by https://github.com/redis/rueidis/pull/472#issuecomment-1952839914

**Pull request closed without merging**